smt/pkg/smt/smt_utils.go

```diff
@@ -0,0 +1,49 @@
+package smt
+
+import (
+	"fmt"
+
+	"github.com/ledgerwatch/erigon/smt/pkg/utils"
+)
+
+var (
+	ErrEmptySearchPath = fmt.Errorf("search path is empty")
+)
+
+func (s *SMT) GetNodeAtPath(path []int) (nodeV *utils.NodeValue12, err error) {
+	pathLen := len(path)
+	if pathLen == 0 {
+		return nil, ErrEmptySearchPath
+	}
+
+	var sl utils.NodeValue12
+
+	oldRoot, err := s.getLastRoot()
+	if err != nil {
+		return nil, fmt.Errorf("getLastRoot: %w", err)
+	}
+
+	for level, pathByte := range path {
+		sl, err = s.Db.Get(oldRoot)
```

Comment:
Does the old change while inside this for ?

Reply:
Yes, it is on line 41

Reply:
but we don't save it in the DB.
do we have to make a DB connection to fetch it every time instead of keeping it saved outside by having 2 oldroot ?

staticOldRoot = static call.
changingOldRoot = something

Reply:
Understood